src/routes/manageroles.js

```diff
@@ -0,0 +1,11 @@
+const router = require('express').Router();
+const { permissionMiddleware } = require('../middlewares/permission');
+const ManageRolesController = require('../controllers/manageroles');
+
+router.post('/create', permissionMiddleware([]), ManageRolesController.PostRole);
+router.get('/list', permissionMiddleware([]), ManageRolesController.ListRoles);
+router.patch('/:id', permissionMiddleware([]), ManageRolesController.UpdateRole);
+router.delete('/:id', permissionMiddleware([]), ManageRolesController.DeleteRole);
+router.get('/:id', permissionMiddleware([]), ManageRolesController.GetRole);
+
```

Comment:
Some endpoints which were specified in documentation were not created:
- Create a role by id (not actually required to implement, but remove documentation)
- List all permissions(This is actually required to create a role: documentation specified that it should come under `/api/permissions/all` but put in under `/api/permissions/roles`)
- Authentication levels required for each request were not mentioned in documentation

Reply:
/api/permissions/roles or /api/roles/permissions?

Reply:
/api/permissions/roles or /api/roles/permissions?